### ISSUE TITLE
Unhandled exception on latest Ubuntu 18.04.1 because of upgrade to libcurl4

### ISSUE THREAD

**Issue by adamskt** [terrajobst/git-istage#13]
Running dotnet CLI 2.1.5 on Ubuntu 18.04.1 LTS, git-istage throws this exception:
```
Unhandled Exception: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library 'git2-8e0b172' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libgit2-8e0b172: cannot open shared object file: No such file or directory
   at LibGit2Sharp.Core.NativeMethods.git_libgit2_init()
   at LibGit2Sharp.Core.NativeMethods.LoadNativeLibrary()
   at LibGit2Sharp.Core.NativeMethods..cctor()
   --- End of inner exception stack trace ---
   at LibGit2Sharp.Core.NativeMethods.git_repository_discover(GitBuf buf, FilePath start_path, Boolean across_fs, FilePath ceiling_dirs)
   at LibGit2Sharp.Core.Proxy.ConvertPath(Func`2 pathRetriever)
   at LibGit2Sharp.Repository.Discover(String startingPath)
   at GitIStage.Program.ResolveRepositoryPath() in /home/vsts/work/1/s/src/git-istage/Program.cs:line 36
   at GitIStage.Program.Main() in /home/vsts/work/1/s/src/git-istage/Program.cs:line 14
error: git-istage died of signal 6
```
Turning up LD_DEBUG yields this nugget:
```
7259:	/usr/lib/x86_64-linux-gnu/libcurl.so.4: error: version lookup error: version `CURL_OPENSSL_3' not found (required by /home/thor/.dotnet/tools/.store/git-istage/0.2.2/git-istage/0.2.2/tools/netcoreapp2.1/any/runtimes/linux-x64/native/libgit2-8e0b172.so) (fatal)
```

It looks like LibGit2Sharp depends on libcurl3 as referenced in [this issue](https://github.com/libgit2/libgit2sharp/issues/1585).  So this will probably go away when LibGit2Sharp updates their native library.

For completion's sake, my dotnet config:
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.403
 Commit:    04e15494b6

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.403/

Host (useful for support):
  Version: 2.1.5
  Commit:  290303f510

.NET Core SDKs installed:
  2.1.403 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by terrajobst** [comment]
AWESOME. Now, these are great bug reports!

@ethomson, what's the best way to make LibGitSharp2 workable for me under Ubuntu 18.04.1? It should be pretty popular for our crowd due to WSL support.

**Comment by ethomson** [comment]
I’m working on reducing the native dependencies that libgit2sharp takes.  In the meantime you can try the experimental builds of that (https://github.com/libgit2/libgit2sharp/pull/1618) or continue to require users to install libcurl3.

**Comment by bording** [comment]
@terrajobst If you don't want to wait for the work @ethomson mentioned that will remove the curl and OpenSSL dependencies completely, the latest preview release, 0.26.0-preview-0062, now includes an `ubuntu.18.04-x64` binary, so this should work with that release.

**Comment by ethomson** [comment]
In addition, I've released a prerelease LibGit2Sharp that should help a bit here.  I've updated libgit2 to have built-in support for proxies, so we no longer rely upon libcurl.  I think this should help the native dependency problems quite a bit.

This is available in [LibGit2Sharp 0.26.0-preview-0070](https://www.nuget.org/packages/LibGit2Sharp/0.26.0-preview-0070).

**Comment by terrajobst** [comment]
I've pushed an update to nuget.org (git-istage 0.2.61). Will take a few minutes before it's available.
